src/Support/Validation/Constraints/WhereConstraint.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Spatie\LaravelData\Support\Validation\Constraints;
+
+class WhereConstraint implements DatabaseConstraint
+{
+    public function __construct(
+        public readonly mixed $column,
+        public readonly mixed $value = null,
```

Comment:
I know indeed laravel has a mixed type but they provide typehints, let's use those. For value mixed is allright since it probably is really mixed. Another thing I would like to see is support for ExternalReference since we can dynamically replace the value which might be useful.

Reply:
Good suggestion! I've updated the PR to allow the use of `ExternalReference`. I also added types to the database constraints to the ones that I found made sense.

I took the type suggestions from https://api.laravel.com/docs/11.x/Illuminate/Validation/Rules/DatabaseRule.html 

I kept the `$value` of the `WhereConstraint` and `WhereNotConstraint` as `mixed`, like you said they probably really are mixed.

The type suggestion for `whereNot` does not include `null`, but that is supported.

This would otherwise error:
```php
new WhereNotConstraint('deleted_at', null)
```
Which is not a big deal, since you can use `WhereNullConstraint` instead, but it makes me think there are other inputs which are not in the type but are supported